index.html

```diff
@@ -153,6 +153,7 @@ <h2>
           readonly attribute GamepadMappingType mapping;
           readonly attribute FrozenArray&lt;double&gt; axes;
           readonly attribute FrozenArray&lt;GamepadButton&gt; buttons;
+          [SameObject] readonly attribute GamepadHapticActuator? vibrationActuator;
```

Comment:
Why is it nullable? that seems a bit odd?

Reply:
The Chromium implementation made it nullable for feature detection. Null `vibrationActuator` means no vibration capability.

It's odd that we now have two levels of feature detection and thus two ways to represent a gamepad with no vibration capability. To make feature detection simpler, the spec should pick one representation.

We should avoid breaking existing apps that use the null-check for feature detection. Making `vibrationActuator` non-nullable would cause these apps to play effects on gamepads that don't support any effects. That might be okay depending on how unsupported effects are handled.

According to the spec, `playEffect` called with an unsupported effect type returns a promise rejected with NotSupportedError. The actual behavior in the Chromium implementation is to resolve the promise with "not-supported". My guess is applications aren't checking for either of these. In the no-vibration-capability case, unexpectedly rejecting with NotSupportedError is probably more disruptive to existing applications than unexpectedly resolving with "not-supported". I think it would be reasonable to make `vibrationActuator` non-nullable if we also change `playEffect` to resolve to "not-supported" in this case.

If we decide to keep `vibrationActuator` nullable then the spec should require it to be null whenever `effects` would be empty.

Reply:
I guess put differently, Chrome could continue to support null, but maybe we shouldn't codify that in the actual spec?

Reply:
I will make the `vibrationActuator` non-nullable in the next commit.

Reply:
I have made it non-nullable